src/Orleans.Core/CodeGeneration/GenericMethodInvoker.cs

```diff
@@ -153,66 +192,121 @@ private static MethodInfo GetTaskConversionMethod(Type taskType)
-            if (obj == null || obj.Length == 0) return 0;
+            if (obj.Length == 0) return 0;
+
             unchecked
             {
-                // Only consider the type parameters.
                 var result = 0;
-                for (var i = 0; i < this.typeParameterCount && i < obj.Length; i++)
+                foreach (var type in obj)
                 {
-                    var type = obj[i] as Type;
-                    if (type == null) break;
-
                     result = (result * 367) ^ type.GetHashCode();
                 }
 
                 return result;
             }
         }
-        
+
         /// <summary>
         /// Returns the <see cref="MethodInfo"/> for the method on <paramref name="declaringType"/> with the provided name
         /// and number of generic type parameters.
         /// </summary>
         /// <param name="declaringType">The type which the method is declared on.</param>
         /// <param name="methodName">The method name.</param>
-        /// <param name="typeParameterCount">The number of generic type parameters.</param>
+        /// <param name="typeParameters">The generic type parameters to use.</param>
+        /// <param name="parameterTypes"></param>
         /// <returns>The identified method.</returns>
-        private static MethodInfo GetMethod(Type declaringType, string methodName, int typeParameterCount)
+        private static MethodInfo GetMethod(
+            Type declaringType,
+            string methodName,
+            Type[] typeParameters,
+            Type[] parameterTypes
+        )
         {
+            MethodInfo methodInfo = null;
+
+            var typeParameterCount = typeParameters.Length;
+
+            bool skipMethod = false;
+
             var methods = declaringType.GetMethods(BindingFlags.Instance | BindingFlags.Public);
-            foreach (var method in methods)
+            foreach (var openMethod in methods)
             {
-                if (!method.IsGenericMethodDefinition) continue;
-                if (!string.Equals(method.Name, methodName, StringComparison.Ordinal)) continue;
-                if (method.GetGenericArguments().Length != typeParameterCount) continue;
+                if (!openMethod.IsGenericMethodDefinition) continue;
+
+                // same name?
+                if (!string.Equals(openMethod.Name, methodName, StringComparison.Ordinal)) continue;
+
+                // same type parameter count?
+                if (openMethod.GetGenericArguments().Length != typeParameterCount) continue;
+
+                MethodInfo closedMethod = null;
+                try
+                {
+                    // fill in type parameters
+                    // NOTE: this may throw due to generic constraints
+                    // https://github.com/dotnet/runtime/issues/28033
+                    closedMethod = openMethod.MakeGenericMethod(typeParameters);
+                }
+                catch (ArgumentException ex)
```

Comment:
Now that I think of it, this will never throw - rosyln has already parsed the original call... right?

Reply:
removed unnecessary try/catch